doc/RELEASE_WALKTHROUGH.rst.txt

```diff
@@ -0,0 +1,139 @@
+
+    $ git checkout maintenance/1.11.x
+    $ git pull upstream maintenance/1.11.x
+    $ git submodule update
+    $ git clean -xdf
+
+Look at the git log to get the hash of the last commit in the release, then
+check it out::
+
+    $ git log
+    $ git checkout e57701a40
+
+Edit pavement.py and setup.py as detailed in HOWTO_RELEASE::
+
+    $ gvim pavement.py setup.py
+    $ git commit -a -m"REL: NumPy 1.11.2rc1 release."
+
+Sanity check::
+
+    $ python runtests.py -m "full"
+    $ python3 runtests.py -m "full"
+
+Tag it, install, and build the source distribution files for sourceforge. The
+install is needed for the documentation which is built using the current
+installed numpy::
+
+    $ git tag -s v1.11.2rc1
+    $ python setup.py install --user
+    $ paver sdist
+    $ paver write_release_and_log
+    $ paver pdf
+
+Check that the documentation files have the correct version on the title page.
+If you skipped the install step above that will probably not be the case. Any pdf
+reader will do::
+
+    $ evince build_doc/pdf/user-guide.pdf
+    $ evince build_doc/pdf/reference.pdf
+
+Sourceforge needs a signed README file::
+
+    $ mv release/NOTES.txt release/README.txt
+    $ gpg -a --clearsign release/README.txt
+    $ mv release/README.txt.asc release/README.txt
+
+The Changelog often has commit messages with long lines and other oddities that
+sneaked by the review process. Clean it up a bit if needed::
+
+    $ gvim release/Changelog
+
+Tag the release branch. Generation of the wheels for PyPI needs this::
+
+    $ git push upstream v1.11.2rc1
+
+Trigger the wheels build. This can take a while. The numpy-wheels.git/
+repository is cloned from `<https://github.com/MacPython/numpy-wheels>`_. Start
+with a pull as the file may have been accessed and changed by someone else and
+a push will fail. The .travis.yml and appveyor.yml files need to be edited to
+make sure they have the correct version::
+
+    $ cd ../numpy-wheels.git/
+    $ git pull origin master
+    $ gvim .travis.yml appveyor.yml
+    $ git commit -a
+    $ git push origin HEAD
+
+Upload to sourceforge
+---------------------
+
+Uploading to sourceforge is easily done online. Log in, make a new directory
+for the upcoming release, and upload the following six source release files.
+
+- build_doc/pdf/reference.pdf
+- build_doc/pdf/userguide.pdf
+- release/installers/numpy-1.11.2rc1.tar.gz
+- release/installers/numpy-1.11.2rc1.zip
+- release/Changelog
+- release/README.txt
+
+
+Upload to PyPI
+--------------
+
+Uploading to PyPI can be a hassle due to the site itself having problems. If
+you get strange messages, put it off for the next day. To begin, check if all
+the needed wheels have been build. There should be 17 of them at
+`<https://wheels.scipy.org>`_, 3 for Mac, 6 for Windows, and 8 for Linux. Here
+the wheelhouse directory is in the home directory. Then upload the
+wheels. You will need to sign every file separately in the process::
+
+    $ cd ../terryfy.git/
```

Comment:
I append `.git` to git repo directories. I can change that for this documentation. This walkthrough probably needs the layout it is written for documented at the beginning.